service/controller/v22/resource/s3bucket/delete.go

```diff
@@ -8,10 +8,22 @@ import (
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/controller"
+	"github.com/giantswarm/operatorkit/controller/context/finalizerskeptcontext"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	"golang.org/x/sync/errgroup"
 
 	"github.com/giantswarm/aws-operator/service/controller/v22/controllercontext"
 )
 
+const (
+	// loopLimit is the maximum amount of delete actions we want to allow per
```

Comment:
Actually this also depends on a number of buckets if I understand the code correctly.